### PR TITLE
Add opengraph meta tags to add details to social media previews

### LINKED ISSUE #226: SEO optimization for https://stats.jenkins.io/
### What feature do you want to see added?

In social media previews, we would like to show the Jenkins logo, a title, as well as a description for https://stats.jenkins.io/. 

### Upstream changes

_No response_

### PR DESCRIPTION
**Pull request by KitsuneKode**
Closes #226 

Added opengraph metadata support to allow social media preview. 

@krisstern 
You can change the description and title according to what you see fit.

### CODE REVIEW

**Comment by KitsuneKode** [comment]
is the preview of not available?

**Comment by krisstern** [comment]
Hi @KitsuneKode Preview is only activated by certain persons with the permissions

File: index.html [hunk 1]
```diff
@@ -2,12 +2,27 @@
...
+        <meta property="og:image" content="https://www.jenkins.io/images/logos/jenkins/jenkins.svg" />
+        <meta property="og:url" content="https://stats.jenkins.io" />
+        <meta name="twitter:card" content="summary_large_image" />
+        <meta property="og:description" content="This site shows different Statistical data of different plugins in the Jenkins Ecosystem " />
+        <meta property="og:site_name" content="Jenkins Statistics" />
+        <meta name="twitter:image:alt" content="Jenkins Statistics Page" />
+
+        <!--  Non-Essential, But Required for Analytics,  -->
+        <meta property="fb:app_id" content="your_app_id" />
+        <meta name="twitter:site" content="@twitter-handle" />
```
Comment (krisstern) [inline index.html:22]: Please comment out for now, better than using placeholder values in production

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by KitsuneKode** [comment]
> Hi @KitsuneKode Preview is only activated by certain persons with the permissions

actually i wanted to check if the metadata unfurls as expected with the preview link,

**Comment by KitsuneKode** [comment]
image

@krisstern this looks fine

**Comment by krisstern** [comment]
@KitsuneKode It would be better if we have a separate `og` image that shows the entirety of Mr. Jenkins instead of only a partt of it. Right now the image is cropped and does not show the top or bottom.

**Comment by krisstern** [comment]
Hi @KitsuneKode I think we can use this one for the `og` image: https://github.com/jenkins-infra/jenkins.io/blob/master/content/images/gsoc/opengraph.png

**Comment by KitsuneKode** [comment]
image

@krisstern   looks good to me

File: index.html [hunk 2]
```diff
@@ -2,12 +2,27 @@
...
+        <link href="/favicon-32x32.png" rel="icon" sizes="32x32" type="image/png" />
+        <link href="/favicon-16x16.png" rel="icon" sizes="16x16" type="image/png" />
         <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+
+        <meta property="og:title" content="Jenkins Statistics" />
+        <meta property="og:type" content="website"/>
+        <meta property="og:image" content="https://raw.githubusercontent.com/jenkins-infra/jenkins.io/refs/heads/master/content/images/gsoc/opengraph.png" />
+        <meta property="og:url" content="https://stats.jenkins.io" />
+        <meta name="twitter:card" content="summary_large_image" />
+        <meta property="og:description" content="This site shows different Statistical data of different plugins in the Jenkins Ecosystem " />
```
Comment (krisstern) [inline index.html:16]:
```suggestion
        <meta property="og:description" content="This site shows statistics related to Jenkins infrastructure " />
```

File: index.html [hunk 3]
```diff
@@ -2,12 +2,27 @@
...
+        <meta property="og:site_name" content="Jenkins Statistics" />
+        <meta name="twitter:image:alt" content="Jenkins Statistics Page" />
```
Comment (krisstern) [inline index.html:18]:
```suggestion
        <meta name="twitter:image:alt" content="Jenkins Statistics" />
```

**Review by krisstern — CHANGES_REQUESTED** [review]

File: index.html (hunk 3 above)
Comment (KitsuneKode) [reply index.html]: done

File: index.html [hunk 4]
```diff
@@ -2,12 +2,27 @@
...
+        <meta property="og:title" content="Jenkins Statistics" />
+        <meta property="og:type" content="website"/>
+        <meta property="og:image" content="https://raw.githubusercontent.com/jenkins-infra/jenkins.io/refs/heads/master/content/images/gsoc/opengraph.png" />
+        <meta property="og:url" content="https://stats.jenkins.io" />
+        <meta name="twitter:card" content="summary_large_image" />
+        <meta property="og:description" content="This site shows statistics related to Jenkins infrastructure" />
+        <meta property="og:site_name" content="Jenkins Statistics" />
+        <meta name="twitter:image:alt" content="Jenkins Statistics" />
+
+        <!--  Non-Essential, But Required for Analytics,  -->
```
Comment (krisstern) [inline index.html:20]:
```suggestion
        <!--  Non-essential, but required for analytics  -->
```

File: index.html [hunk 5]
```diff
@@ -2,12 +2,27 @@
 <html lang="en">
     <head>
         <meta charset="UTF-8" />
-        <link href='/favicon.ico' rel='shortcut icon' type='image/x-icon'>
-        <link href='/apple-touch-icon.png' rel='apple-touch-icon' sizes='180x180'>
-        <link href='/favicon-32x32.png' rel='icon' sizes='32x32' type='image/png'>
-        <link href='/favicon-16x16.png' rel='icon' sizes='16x16' type='image/png'>
+        <link href="/favicon.ico" rel="shortcut icon" type="image/x-icon" />
```
Comment (krisstern) [inline index.html:5]:
```suggestion
        <link href="/favicon.ico" rel="icon" type="image/x-icon" />
```

Comment (krisstern) [reply index.html]: I don't think we use "shortcut" in `rel` for favicon anymore. It used to be a thing though.

Comment (KitsuneKode) [reply index.html]: I didn't know that, learned something new today. I thought it was maybe some there for other reasons

**Review by krisstern — APPROVED** [review]
LGTM! Thanks so much for this PR @KitsuneKode

**Comment by KitsuneKode** [comment]
@krisstern happy to help, got to learn a lot as well

File: index.html (hunk 5 above)
Comment (krisstern) [reply index.html]: Actually I learned about this recently as well

**Pull request merged**